contracts/MemeBase.sol

```diff
@@ -157,34 +158,41 @@ contract MemeBase is MemeFactory {
     function _redemptionSetup(address[] memory accounts, uint256[] memory amounts) private {
         require(accounts.length == amounts.length);
 
-        redemptionAddress = address(new Meme("Agent Token", "AGENT", DECIMALS, MIN_TOTAL_SUPPLY));
+        redemptionAddress = address(new Meme("Agent Token II", "AGNT II", DECIMALS, 1_000_000_000 ether));
 
         // Record all the accounts and amounts
         uint256 totalAmount;
         for (uint256 i = 0; i < accounts.length; ++i) {
-            // Adjust amount for already collected burned tokens
-            uint256 adjustedAmount = (amounts[i] * 9) / 10;
-            totalAmount += adjustedAmount;
-            memeHearters[redemptionAddress][accounts[i]] = adjustedAmount;
+            totalAmount += amounts[i];
+            memeHearters[redemptionAddress][accounts[i]] = amounts[i];
+            // to match original hearter events
+            emit Hearted(accounts[i], redemptionAddress, amounts[i]);
```

Comment:
Our structs and events should be identical to the original ones. Otherwise we introduce potential for confusion